pkg/connectors/protos/register_asset_request.proto

```diff
@@ -0,0 +1,18 @@
+// Copyright 2020 IBM Corp.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package connectors;
+option java_package = "com.datmesh";
+option go_package = "github.com/ibm/the-mesh-for-data/pkg/connectors/protobuf";
+
+import "credentials.proto";
+import "data_store.proto";
+
+message RegisterAssetRequest {
+    string dataset_id = 1;           // identifier of the bucket created for registering asset
+	Credentials creds = 2;
+	DataStore data_store = 3; 	
+	string destination_catalog_id = 4;
+}
```

Comment:
Where is the metadata associated with the new data set?  For example, when ingesting the data owner may provide information such as the data residency, level of confidentiality.